app/models/user.rb

```diff
@@ -220,7 +221,16 @@ def can_downvote?(obj)
   end
 
   def can_invite?
-    !banned_from_inviting? && self.can_submit_stories?
+    closed_testing = Rails.application.closed_testing?
+    max_invitations_count = Rails.application.max_invitations_count
+
+    if self.is_admin || self.is_moderator
```

Comment:
questa si potrebbe semplificare in una sola operazione booleana, ma in effetti è già un po' intricata così com'è

Reply:
Sì l'ho fatto perché altrimenti diventava molto poco leggibile. @PizzolatoDavide tu cosa ne pensi?

Reply:
Io lo riscriverei così.
Alla fine non è un metodo usato in modo frequentissimo quindi preferirei la leggibilità alla velocità di esecuzione.
```
def can_invite?
    if self.is_admin || self.is_moderator
        return true
    else
        if !banned_from_inviting? && self.can_submit_stories?
            closed_testing = Rails.application.closed_testing?
            max_invitations_count = Rails.application.max_invitations_count
            return !closed_testing || return self.invitations_sent_count < max_invitations_count
        else
            return false
        end
     end
end
```

Reply:
Sono d'accordo sull'idea di renderlo il più leggibile possibile. L'unica cosa che non mi piace del tuo codice è `return !closed_testing || return self.invitations_sent_count < max_invitations_count`. Io lo capisco meglio scritto così: `return closed_testing ? self.invitations_sent_count < max_invitations_count : true` però forse è solo una cosa mia.

Sentiamo da @chobeat.

Reply:
Secondo me l'originale era più leggibile

Reply:
Sì dai lasciamo l'originale. Magari sposto
```
closed_testing = Rails.application.closed_testing?
max_invitations_count = Rails.application.max_invitations_count
```
dentro all'`else` in modo che vengano eseguiti solo se non è un admin o moderatore.

Reply:
ok